### PR TITLE
Add trust core files

### PR DESCRIPTION
**Pull request by alejandro-isaza**
This PR moves most source files from Trust Core.

### CODE REVIEW

File: swift/Sources/Blockchains/BlockchainFactory.swift [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+    case .bitcoinCash:
+        return BitcoinCash()
+    case .tomoChain:
+        return TomoChain()
+    case .zcoin:
+        return Zcoin()
+    case .binance:
+        return BinanceChainTestnet()
+    case .ripple:
+        fatalError("not implemented")
```
Comment (hewigovens) [inline swift/Sources/Blockchains/BlockchainFactory.swift]: `Ripple` is implemented

File: swift/Sources/Addresses/BitcoinAddressExtension.swift [hunk 2]
```diff
@@ -0,0 +1,25 @@
+// Copyright © 2017-2019 Trust.
+//
+// This file is part of Trust. The full Trust copyright notice, including
+// terms governing use, modification, and redistribution, is contained in the
+// file LICENSE at the root of the source code distribution tree.
+
+import Foundation
+
+extension BitcoinAddress: Address, Equatable {
```
Comment (hewigovens) [inline swift/Sources/Addresses/BitcoinAddressExtension.swift]: Filename suggestion: `BitcoinAddress+Extension.swift`

Comment (alejandro-isaza) [reply swift/Sources/Addresses/BitcoinAddressExtension.swift]: I like that pattern when the part after the `+` describes what the extension is for.  In this case it doesn't.

File: swift/Sources/Blockchains/BlockchainFactory.swift (hunk 1 above)
Comment (alejandro-isaza) [reply swift/Sources/Blockchains/BlockchainFactory.swift]: Good catch.

File: swift/Tests/Blockchains/DashTests.swift [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+    func testAddress() {
+        let privateKey = PrivateKey(wif: "XDoxFwfxsEZDd15uNyj8vt64c3GLxcFjTefnUz7gckvAJeYSFaRz")!
+        let publicKey = privateKey.getPublicKey(compressed: true)
+        let address = Dash().address(for: publicKey)
+
+        XCTAssertEqual(address.description, "Xw7HTXGY3TFeA3ZsVuMRrYh96GtwWb4hQb")
+    }
+
+    func testExtendedKeys() {
+        let wallet = HDWallet(mnemonic: "ripple scissors kick mammal hire column oak again sun offer wealth tomorrow wagon turn fatal", passphrase: "TREZOR")
```
Comment (vikmeup) [inline swift/Tests/Blockchains/DashTests.swift]: `ripple scissors kick mammal hire column oak again sun offer wealth tomorrow wagon turn fatal` is used in 4 places, can we create one class for it?

Comment (hewigovens) [reply swift/Tests/Blockchains/DashTests.swift]: we can create a factory method for that

Comment (vikmeup) [reply swift/Tests/Blockchains/DashTests.swift]: 👍

**Pull request merged**